src/OpenLoco/src/GameCommands/CreateIndustry.cpp

```diff
@@ -0,0 +1,722 @@
+        RangeAndMinDistance{ 3, -1 },
+        RangeAndMinDistance{ 3, -1 },
+        RangeAndMinDistance{ 3, -1 },
+        RangeAndMinDistance{ 3, -1 },
+        RangeAndMinDistance{ 3, -1 }, // 7
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+        RangeAndMinDistance{ 5, -2 },
+    };
+
+    // 0x0045436B
+    static currency32_t createIndustry(const IndustryPlacementArgs& args, const uint8_t flags)
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Miscellaneous);
+        {
+            const auto centrePos = World::Pos2(args.pos.x + 16, args.pos.y + 16);
+            const auto centreHeight = World::TileManager::getHeight(centrePos);
+            GameCommands::setPosition({ args.pos.x, args.pos.y, centreHeight.landHeight });
+        }
+
+        // Vanilla bug args.srand0 not set. Change this when we can diverge
+        Core::Prng prng{ enumValue(GameCommand::createIndustry), args.srand1 };
+        const auto newIndustryId = sub_454C91(args.type, args.pos, prng);
+        if (newIndustryId == IndustryId::null)
+        {
+            return FAILURE;
+        }
+        _industryLastPlacedId = newIndustryId;
+        auto* newIndustry = IndustryManager::get(newIndustryId);
+        auto* indObj = newIndustry->getObject();
+        if (args.buildImmediately)
+        {
+            newIndustry->under_construction = 0xFF;
+        }
+        if (flags & Flags::ghost)
+        {
+            newIndustry->flags |= IndustryFlags::isGhost;
+        }
+        if (!isEditorMode() && getUpdatingCompanyId() != CompanyId::neutral)
+        {
+            newIndustry->flags |= IndustryFlags::flag_04;
+            newIndustry->owner = getUpdatingCompanyId();
+        }
+
+        // 0x00E0C3D3
+        const auto randColour = getRandomAvailableColour(newIndustry->prng, *indObj);
+
+        // 0x00E0C3BE - C0
+        auto lastPlacedBuildingPos = World::Pos2{ newIndustry->x, newIndustry->y };
+
+        // used also for 0x00454552 break up into two when function allowed to diverge
+        const auto randVal = newIndustry->prng.randNext() & 0xFF;
+
+        auto prodRateRand = randVal;
+        for (auto i = 0; i < 2; ++i)
+        {
+            newIndustry->var_17D[i] = 0;
+            newIndustry->productionRate[i] = (((indObj->initialProductionRate[i].max - indObj->initialProductionRate[i].min) * prodRateRand) / 256) + indObj->initialProductionRate[i].min;
```

Comment:
Could we introduce a const temporary for `indObj->initialProductionRate[i]` for legibility?